domain/secret/service/delete.go

```diff
@@ -13,25 +13,28 @@ import (
 	"github.com/juju/juju/internal/secrets/provider"
 )
 
-func (s *SecretService) DeleteCharmSecret(ctx context.Context, uri *secrets.URI, revisions []int, canDelete func(uri *secrets.URI) error) error {
-	return nil
-}
-
 // DeleteObsoleteUserSecrets deletes any obsolete user secret revisions that are marked as auto-prune.
 func (s *SecretService) DeleteObsoleteUserSecrets(ctx context.Context) error {
 	// TODO(secrets)
 	return nil
 }
 
-// DeleteUserSecret removes the specified user supplied secret.
-// The secret is removed from state and backend.
-func (s *SecretService) DeleteUserSecret(ctx context.Context, uri *secrets.URI, revisions []int) error {
-	// TODO(secrets) - get model uuid from state
-	var modelUUID string
+// DeleteSecret removes the specified secret.
+// If revisions is nil or the last remaining revisions are removed.
+// It returns [secreterrors.PermissionDenied] if the secret cannot be managed by the accessor.
+func (s *SecretService) DeleteSecret(ctx context.Context, uri *secrets.URI, params DeleteSecretParams) error {
```

Comment:
Is this method for user secret or charm secret?
Line 49 uses model UUID as owner, I guess it's for user secret?
So we need a better method name.

Reply:
It's delete for any type of secret, just like update is for any type of secret.

Line 49 is wrong and needs to be fixed when the backend support is added. Tags will be removed and logic reworked. For now, secret backends not supported. There's a bit of backend code just copied down from the facade as a placeholder in various methods. I can add a TODO